### PR TITLE
refactor: normalize logical operator arguments while parsing

### PR DESCRIPTION
**Pull request by cyjake**

### CODE REVIEW

File: src/drivers/abstract/spellbook.js [hunk 1]
```diff
@@ -574,7 +571,7 @@ function formatDelete(spell) {
 function formatConditions(spell, conditions) {
   return conditions
     .map(condition => {
-      return isLogicalOp(condition) && condition.name == 'or' && conditions.length > 1 && condition.args.length > 1
+      return isLogicalOp(condition) && condition.name == 'or' && conditions.length > 1
```
Comment (cyjake) [inline src/drivers/abstract/spellbook.js:574]: spellbook.js reverted

Comment (cyjake) [reply src/drivers/abstract/spellbook.js:574]: oh, btw, if we intend to start abstracting SQL manipulation code, I think starting with spellbook is a good idea.

File: src/spell.js [hunk 2]
```diff
@@ -148,38 +148,10 @@ function parseLogicalObjectConditionValue(value) {
...
- *     $not: [
- *       'Leah',
- *       'jss'
- *     ]
- *   }
- * }
+ * { $or: [ { title: 'Leah' }, { content: 'Diablo' } ] }
+ * { title: { $or: [ 'Leah', 'Diablo' ] } }
+ * { title: { $or: [ 'Leah', { $like: '%jjj' } ] } }
+ * { title: { $not: [ 'Leah', 'jss' ] } }
```
Comment (cyjake) [inline src/spell.js:154]: changed into a format that is more compact, like the output of `util.inspect(obj)`

File: src/spell.js [hunk 3]
```diff
@@ -200,6 +172,12 @@ function parseLogicalObjectCondition(name, value) {
     return res.concat(arg);
   }, []);
 
+  if (args.length === 0) {
+    throw new Error(`insufficient logical operator value ${value}`);
+  }
+  // malformed logical condition: { title: { $or: [ 'Leah' ] } }
+  if (args.length === 1 && operator !== 'not') return args[0];
```
Comment (cyjake) [inline src/spell.js:179]: $and/$or requires arity at least above two

**Review by JimmyDaddy — APPROVED** [review]

**Pull request merged**